File: truevfs-ext/truevfs-ext-logging/src/main/java/net/java/truevfs/ext/logging/LogBuffer.java
/*
 * Copyright © 2005 - 2021 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging;

import net.java.truecommons.cio.IoBuffer;
import net.java.truevfs.comp.inst.InstrumentingBuffer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;

final class LogBuffer extends InstrumentingBuffer<LogMediator> implements LogResource {

    private static final Logger logger = LoggerFactory.getLogger(LogBuffer.class);

    LogBuffer(LogMediator mediator, IoBuffer entry) {
        super(mediator, entry);
        log("Allocated {}", entry);
    }

    @Override
    public void release() throws IOException {
        log("Releasing {}", entry);
        entry.release();
    }

    @Override
    public Logger logger() {
        return logger;
    }
}

File: truevfs-ext/truevfs-ext-logging/src/main/scala/net/java/truevfs/ext/logging/LogBuffer.scala
/*
 * Copyright (C) 2005-2015 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging

import net.java.truevfs.comp.inst._
import net.java.truecommons.cio._
import org.slf4j._

/**
 * @author Christian Schlichtherle
 */
private final class LogBuffer(director: LogMediator, buffer: IoBuffer)
extends InstrumentingBuffer[LogMediator](director, buffer) with LogResource {

  override def logger = LogBuffer.logger

  log("Allocated {}", entry)

  override def release {
    log("Releasing {}", entry)
    entry.release
  }
}

private object LogBuffer {
  private val logger = LoggerFactory.getLogger(classOf[LogBuffer])
}
